lib/enum/arrays.rb

```diff
@@ -1,24 +1,32 @@
 class Array
 
   def sum_of_numeric_elements
+    # inject(0){ |memo, element| element.is_a? Numeric ? (memo + element) : element }
+    select { |element| element.is_a? Numeric }.inject(:+)
   end
 
   def product_of_numeric_elements
+    select{ |element| element.is_a? Numeric }.inject(:*)
   end
 
   def even_numeric_elements
+    select{ |element| element.is_a?(Fixnum) && (element % 2 == 0)}
```

Comment:
Yes, and Ruby also has a built-in method called .even? (and .odd? for the next problem)